cfgov/cal/tests/test_forms.py

```diff
@@ -0,0 +1,203 @@
+import datetime
+
+from django import forms
+from unittest import TestCase
+
+from cal.forms import (
+    CalendarFilterForm, CalendarPDFForm, FilterCheckboxList, FilterDateField
+)
+from cal.models import CFPBCalendar
+
+
+class FilterCheckboxListTestCase(TestCase):
+    def get_field(self, **kwargs):
+        return FilterCheckboxList(
+            choices=[(c, c) for c in ('a', 'b', 'c')],
+            **kwargs
+        )
+
+    def test_value_not_required(self):
+        self.get_field(required=False).validate('a')
+
+    def test_value_required(self):
+        self.get_field().validate('a')
+
+    def test_no_value_not_required(self):
+        self.get_field(required=False).validate([])
+
+    def test_no_value_required(self):
+        field = self.get_field()
+        self.assertRaises(forms.ValidationError, field.validate, None)
+
+    def test_invalid_value_passes(self):
+        self.get_field().validate('d')
```

Comment:
@sebworks just wanted to call out this test specifically. Currently because the calendar options are hardcoded on the frontend, it's possible that a user might make a query for a calendar even when there's no data on the backend. This happens if you test this code against a local DB without calendar data, for example. The pre-existing behavior is that this search just returns nothing silently, instead of actually failing validation and reporting an error to the user. So my changes to the form code and these tests just keep that existing behavior.